From a483b0d4d6a16c07bcc7f65564974f5fc26c26d8 Mon Sep 17 00:00:00 2001
From: Sebastian Lackner <sebastian@fds-team.de>
Date: Mon, 9 Nov 2015 17:19:42 +0100
Subject: kernel32/tests: Only allow one test result.

This reverts commit 3b7cdc0b8475f4eac910c5260c96c381ddbb630e.
Instead of hiding errors, update the todo_wine flags accordingly.
---
 dlls/kernel32/tests/pipe.c | 24 ++++++------------------
 1 file changed, 6 insertions(+), 18 deletions(-)

diff --git a/dlls/kernel32/tests/pipe.c b/dlls/kernel32/tests/pipe.c
index ae09ef6..62ac77f 100644
--- a/dlls/kernel32/tests/pipe.c
+++ b/dlls/kernel32/tests/pipe.c
@@ -294,14 +294,11 @@ static void test_CreateNamedPipe(int pipemode)
         ok(written == sizeof(obuf2), "write file len 3b\n");
         ok(PeekNamedPipe(hFile, ibuf, sizeof(ibuf), &readden, &avail, NULL), "Peek3\n");
         if (pipemode == PIPE_TYPE_BYTE) {
-            /* currently the Wine behavior depends on the kernel version */
-            /* ok(readden == sizeof(obuf) + sizeof(obuf2), "peek3 got %d bytes\n", readden); */
-            if (readden != sizeof(obuf) + sizeof(obuf2)) todo_wine ok(0, "peek3 got %d bytes\n", readden);
+            ok(readden == sizeof(obuf) + sizeof(obuf2), "peek3 got %d bytes\n", readden);
         }
         else
         {
-            /* ok(readden == sizeof(obuf), "peek3 got %d bytes\n", readden); */
-            if (readden != sizeof(obuf)) todo_wine ok(0, "peek3 got %d bytes\n", readden);
+            todo_wine ok(readden == sizeof(obuf), "peek3 got %d bytes\n", readden);
         }
         ok(avail == sizeof(obuf) + sizeof(obuf2), "peek3 got %d bytes available\n", avail);
         pbuf = ibuf;
@@ -325,14 +322,11 @@ static void test_CreateNamedPipe(int pipemode)
         ok(written == sizeof(obuf2), "write file len 4b\n");
         ok(PeekNamedPipe(hnp, ibuf, sizeof(ibuf), &readden, &avail, NULL), "Peek4\n");
         if (pipemode == PIPE_TYPE_BYTE) {
-            /* currently the Wine behavior depends on the kernel version */
-            /* ok(readden == sizeof(obuf) + sizeof(obuf2), "peek4 got %d bytes\n", readden); */
-            if (readden != sizeof(obuf) + sizeof(obuf2)) todo_wine ok(0, "peek4 got %d bytes\n", readden);
+            ok(readden == sizeof(obuf) + sizeof(obuf2), "peek4 got %d bytes\n", readden);
         }
         else
         {
-            /* ok(readden == sizeof(obuf), "peek4 got %d bytes\n", readden); */
-            if (readden != sizeof(obuf)) todo_wine ok(0, "peek4 got %d bytes\n", readden);
+            todo_wine ok(readden == sizeof(obuf), "peek4 got %d bytes\n", readden);
         }
         ok(avail == sizeof(obuf) + sizeof(obuf2), "peek4 got %d bytes available\n", avail);
         pbuf = ibuf;
@@ -373,10 +367,7 @@ static void test_CreateNamedPipe(int pipemode)
             ok(WriteFile(hnp, obuf2, sizeof(obuf2), &written, NULL), " WriteFile5b\n");
             ok(written == sizeof(obuf2), "write file len 3b\n");
             ok(PeekNamedPipe(hFile, ibuf, sizeof(ibuf), &readden, &avail, NULL), "Peek5\n");
-            /* currently the Wine behavior depends on the kernel version */
-            /* ok(readden == sizeof(obuf), "peek5 got %d bytes\n", readden); */
-            if (readden != sizeof(obuf)) todo_wine ok(0, "peek5 got %d bytes\n", readden);
-
+            todo_wine ok(readden == sizeof(obuf), "peek5 got %d bytes\n", readden);
             ok(avail == sizeof(obuf) + sizeof(obuf2), "peek5 got %d bytes available\n", avail);
             pbuf = ibuf;
             ok(memcmp(obuf, pbuf, sizeof(obuf)) == 0, "content 5a check\n");
@@ -408,10 +399,7 @@ static void test_CreateNamedPipe(int pipemode)
             ok(WriteFile(hFile, obuf2, sizeof(obuf2), &written, NULL), " WriteFile6b\n");
             ok(written == sizeof(obuf2), "write file len 6b\n");
             ok(PeekNamedPipe(hnp, ibuf, sizeof(ibuf), &readden, &avail, NULL), "Peek6\n");
-            /* currently the Wine behavior depends on the kernel version */
-            /* ok(readden == sizeof(obuf), "peek6 got %d bytes\n", readden); */
-            if (readden != sizeof(obuf)) todo_wine ok(0, "peek6 got %d bytes\n", readden);
-
+            todo_wine ok(readden == sizeof(obuf), "peek6 got %d bytes\n", readden);
             ok(avail == sizeof(obuf) + sizeof(obuf2), "peek6b got %d bytes available\n", avail);
             pbuf = ibuf;
             ok(memcmp(obuf, pbuf, sizeof(obuf)) == 0, "content 6a check\n");
-- 
2.6.2

